features/0728-device-binding-attachments/README.md

```diff
@@ -0,0 +1,261 @@
+
+
+The `device-binding-challenge` must be attached to the `request-presentations~attach` array of the `request-presentation` message defined by [RFC-0037](https://github.com/hyperledger/aries-rfcs/blob/main/features/0037-present-proof/README.md#request-presentation) and [RFC-0454](https://github.com/hyperledger/aries-rfcs/tree/main/features/0454-present-proof-v2#request-presentation).
+
+#### Example request-presentation messages
+
+The following represents a request-presentation message with an attached libindy presentation request and a corresponding device-binding-challenge.
+
+**Present Proof v1**
+```json=
+{
+    "@type": "https://didcomm.org/present-proof/1.0/request-presentation",
+    "@id": "<uuid-request>",
+    "comment": "some comment",
+    "request_presentations~attach": [
+        {
+            "@id": "libindy-request-presentation-0",
+            "mime-type": "application/json",
+            "data":  {
+                "base64": "<bytes for base64>"
+            }
+        }
+    ],
+    "device_binding~attach": [
+        {
+            "@id": "device-binding-challenge-0"
+            "mime-type": "application/json",
+            "data":  {
+                "base64": "<device-binding-challenge>"
+            }
+        }
+    ]
+}
+```
+
+**Present Proof v2**
+
+```json=
+{
+    "@type": "https://didcomm.org/present-proof/2.0/request-presentation",
+    "@id": "<uuid-request>",
+    "goal_code": "<goal-code>",
+    "comment": "some comment",
+    "will_confirm": true,
+    "present_multiple": false,
+    "formats" : [
+        {
+            "attach_id" : "libindy-request-presentation-0",
+            "format" : "hlindy/proof-req@v2.0",
+        }
+    ],
+    "request_presentations~attach": [
+        {
+            "@id": "libindy-request-presentation-0",
+            "mime-type": "application/json",
+            "data":  {
+                "base64": "<base64 data>"
+            }
+        }
+    ],
+    "device_binding~attach": [
+        {
+            "@id": "device-binding-challenge-0"
+            "mime-type": "application/json",
+            "data":  {
+                "base64": "<device-binding-challenge>" // inner object
+            }
+        }
+    ]
+}
+```
+
+
+### Response
+
+The following response must be generated by the holder of the VC.
+
+#### device-binding-reponse
+```json=
+{
+    "@type": "https://didcomm.org/device-binding/%ver/device-binding-response",
+    "@id": "<uuid-challenge-response>",
+    "proofs" : [
+        {
+            "id": "libindy-presentation-0",
+            "path": "$.requested_proof.revealed_attrs.attr1_referent.raw",
+            "jws": {
+              "header": {
+                "kid": "did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"
+              },
+              "protected": "eyJhbGciOiJFZERTQSIsImlhdCI6MTU4Mzg4... (bytes omitted)",
+              "signature": "3dZWsuru7QAVFUCtTd0s7uc1peYEijx4eyt5... (bytes omitted)"
+            }
```

Comment:
Would it be possible to make this a signed attachment to reuse the existing functionality?

Reply:
Do you mean to move `jws` up into part of the attachment next to the `data` part of the attachment?

Reply:
Yes! So it follows the spec from RFC 0017

Reply:
Can you check the updated spec?